infrahub_sdk/ctl/menu.py

```diff
@@ -40,21 +44,54 @@ async def load(
     files = load_yamlfile_from_disk_and_exit(paths=menus, file_type=MenuFile, console=console)
     client = initialize_client()
 
+    has_errors = False
+
+    for file in files:
+        try:
+            await file.validate_format(client=client, branch=branch)
+        except ValidationError as exc:
+            has_errors = True
+            display_object_validate_format_error(file=file, error=exc, console=console)
+
+    if has_errors:
+        raise typer.Exit(1)
+
     for file in files:
-        file.validate_content()
-        schema = await client.schema.get(kind=file.spec.kind, branch=branch)
-
-        for idx, item in enumerate(file.spec.data):
-            try:
-                await file.spec.create_node(
-                    client=client,
-                    schema=schema,
-                    position=[idx + 1],
-                    data=item,
-                    branch=branch,
-                    default_schema_kind=file.spec.kind,
-                    context={"list_index": idx},
-                )
-            except ObjectValidationError as exc:
-                console.print(f"[red] {exc!s}")
-                raise typer.Exit(1)
+        try:
+            await file.process(client=client, branch=branch)
+        except ObjectValidationError as exc:
+            has_errors = True
+            console.print(f"[red] {exc!s}")
+
+    if has_errors:
+        raise typer.Exit(1)
+
+
+@app.command()
+@catch_exception(console=console)
+async def validate(
```

Comment:
Would be nice with a unit test even if it's just for the failure scenario

Reply:
Good point, I added some unit tests